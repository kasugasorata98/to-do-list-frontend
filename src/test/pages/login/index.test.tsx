import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Login from "@/pages/login";

describe("Login", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Login />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the login page correctly", () => {
    render(<Login />);

    expect(screen.getByText("Login")).toBeInTheDocument();
    expect(screen.getByText("Please sign in to continue.")).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent("Login with Google");
  });
});
