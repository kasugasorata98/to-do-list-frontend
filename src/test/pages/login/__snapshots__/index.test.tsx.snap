// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ielbyb"
  >
    <h2
      class="chakra-heading css-13mo70e"
    >
      Login
    </h2>
    <p
      class="chakra-text css-0"
      style="color: rgb(226, 234, 241); margin-bottom: 4px;"
    >
      Please sign in to continue.
    </p>
    <button
      class="chakra-button css-4xx2wk"
      style="color: rgb(60, 66, 73);"
      type="button"
    >
      Login with Google
    </button>
  </div>
</DocumentFragment>
`;
